python-lib/spell_checker/spell_checker.py

```diff
@@ -0,0 +1,127 @@
+import dataiku
+from symspellpy.symspellpy import SymSpell, Verbosity
+from spell_checker.language_detection import *
+from spell_checker.dataframe_helpers import *
+from spell_checker.text_preprocessing import *
+
+
+class SpellChecker:
+    def __init__(self, params):
+        self.params = params
+
+        # Root path of frequency dictionaries
+        self.resource_path = dataiku.customrecipe.get_recipe_resource()
```

Comment:
Resource path should be a variable defined outside of the class. As much as possible, we should decorrelate classes from the Dataiku API, so that the code can be ported and tested elsewhere.